pkg/controllers/cassandra/pilot/pilot.go

```diff
@@ -79,28 +74,22 @@ func (c *pilotControl) clusterPods(cluster *v1alpha1.CassandraCluster) ([]*v1.Po
 	return clusterPods, nil
 }
 
-func (c *pilotControl) createOrUpdatePilot(cluster *v1alpha1.CassandraCluster, pod *v1.Pod) error {
+func (c *pilotControl) createPilot(cluster *v1alpha1.CassandraCluster, pod *v1.Pod) error {
 	desiredPilot := PilotForCluster(cluster, pod)
 	client := c.naviClient.NavigatorV1alpha1().Pilots(desiredPilot.GetNamespace())
 	lister := c.pilots.Pilots(desiredPilot.GetNamespace())
-	existingPilot, err := lister.Get(desiredPilot.GetName())
-	if k8sErrors.IsNotFound(err) {
-		_, err = client.Create(desiredPilot)
-		return err
+	_, err := lister.Get(desiredPilot.GetName())
+	// Pilot already exists
+	if err == nil {
+		return nil
 	}
-	if err != nil {
+	// The only error we expect is that the pilot does not exist.
+	if !k8sErrors.IsNotFound(err) {
 		return err
 	}
-	err = util.OwnerCheck(existingPilot, cluster)
-	if err != nil {
-		return err
-	}
-	existingPilot = existingPilot.DeepCopy()
-	existingPilot.Status = v1alpha1.PilotStatus{}
-	desiredPilot = existingPilot.DeepCopy()
-	desiredPilot = updatePilotForCluster(cluster, pod, desiredPilot)
-	if !reflect.DeepEqual(desiredPilot, existingPilot) {
-		_, err = client.Update(desiredPilot)
+	_, err = client.Create(desiredPilot)
+	if k8sErrors.IsAlreadyExists(err) {
+		return nil
```

Comment:
I worry returning nil here just because a pilot with that same name already exists could cause problems. Similar to my comments in #291, I think we should return an error here and let the next invocation of the loop take the correct action (once the lister is consistent)